subprojects/core-api/src/main/java/org/gradle/api/invocation/Gradle.java

```diff
@@ -359,4 +359,11 @@
      */
     @Incubating
     IncludedBuild includedBuild(String name);
+
+    /**
+     * Enables a feature preview by name.
+     *
+     * @param name the name of the feature to enable
+     */
+    void enableFeaturePreview(String name);
```

Comment:
Adding this here makes me wonder about two things:
* I believe we should not allow any changes beyond the configuration phase. Is that correct? And if it is, any pointers on how to do that?
* Some features may have an even stricter lifecycle constraint. For example, advanced pom support should not be turned on if a configuration has already been resolved. Does that make sense?

Reply:
One option we have is to lock these options as soon as the settings are loaded, so that they can only be set in either an init script or a settings script. That would avoid more difficult questions about lifecycle. Locking after configuration is another option, but it makes it difficult to reason about resolution that happens during the configuration phase.

I don't think forcing users to set this in `settings.gradle` (or `init.gradle`) is too big of a limitation.

Reply:
Re-adding my reply, so it's part of the review:
One option we have is to lock these options as soon as the settings are loaded, so that they can only be set in either an init script or a settings script. That would avoid more difficult questions about lifecycle. Locking after configuration is another option, but it makes it difficult to reason about resolution that happens during the configuration phase.

I don't think forcing users to set this in settings.gradle (or init.gradle) is too big of a limitation.

Reply:
Sadly I do not think that is an option, as we have at least one plugin that enables a feature when applied. And my quick review of the [build lifecycle doc](https://docs.gradle.org/current/userguide/build_lifecycle.html) did not indicate another moment that could be used as a fence.